portal-frontend/src/app/features/application-details/application-details.component.ts

```diff
@@ -49,9 +49,9 @@ export class ApplicationDetailsComponent implements OnInit, OnDestroy {
   ) {}
 
   ngOnInit(): void {
-    this.loadGovernments();
     this.$application.pipe(takeUntil(this.$destroy)).subscribe((app) => {
       this.applicationSubmission = app;
+      this.loadGovernments();
```

Comment:
this will reload LG every time application is reloaded, is it expected?

Reply:
You're right let me rethink this approach. Thanks Mekhti!